src/main/java/mate/academy/EventManager.java

```diff
@@ -1,19 +1,38 @@
 package mate.academy;
 
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CopyOnWriteArrayList;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
 public class EventManager {
+    private ExecutorService executor;
+    private List<EventListener> listeners = new CopyOnWriteArrayList<>();
+
+    public EventManager() {
+        executor = Executors.newFixedThreadPool(5);
+    }
+
     public void registerListener(EventListener listener) {
+        listeners.add(listener);
 
     }
 
     public void deregisterListener(EventListener listener) {
-
+        listeners.remove(listener);
     }
 
     public void notifyEvent(Event event) {
-
+        for (EventListener listener : listeners) {
+            listener.onEvent(event);
+        }
+        for (EventListener listener : listeners) {
+            CompletableFuture.runAsync(listener::getProcessedEvent, executor);
```

Comment:
Using `CompletableFuture.runAsync` without handling exceptions can lead to unobserved exceptions if the `getProcessedEvent` method throws an error. It's important to handle exceptions, possibly by using a try-catch block within the `runAsync` lambda or by chaining a `handle` method to the `CompletableFuture`.